From 9507b6abd557d5a446bee9cc0dfd3897124822ea Mon Sep 17 00:00:00 2001
Message-Id: <9507b6abd557d5a446bee9cc0dfd3897124822ea.1374136911.git.mkletzan@redhat.com>
In-Reply-To: <1511d0e8f43bbff718747c8e4c0cb79892bd2dc1.1374136911.git.mkletzan@redhat.com>
References: <1511d0e8f43bbff718747c8e4c0cb79892bd2dc1.1374136911.git.mkletzan@redhat.com>
From: Martin Kletzander <mkletzan@redhat.com>
Date: Mon, 22 Apr 2013 16:54:57 +0200
Subject: [RHEL-6.5 python-virtinst PATCH 3/5] Support incomplete .treeinfo
 files

In case there was '.treeinfo' file available, virt-install was
forcibly getting image information out of the file.  For some
distributions (especially older releases, e.g. RHEL 4), there might be
'.treeinfo' file available, but with incomplete information.  Allow
fallback to default values even when '.treeinfo' file is found.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=954262
(cherry picked from commit 70fa76a78037e939d975b3977564b89e76f799a4)
(cherry picked from commit 1ca4fc1945c6aefe0d97b21b2f0f2bffc14bb3db)
---
 virtinst/OSDistro.py | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/virtinst/OSDistro.py b/virtinst/OSDistro.py
index f900f66..9ff1814 100644
--- a/virtinst/OSDistro.py
+++ b/virtinst/OSDistro.py
@@ -1,7 +1,7 @@
 #
 # Represents OS distribution specific install data
 #
-# Copyright 2006-2007  Red Hat, Inc.
+# Copyright 2006-2007, 2013  Red Hat, Inc.
 # Daniel P. Berrange <berrange@redhat.com>
 #
 # This program is free software; you can redistribute it and/or modify
@@ -257,9 +257,13 @@ class Distro:
         kernelpath = None
         initrdpath = None
         if self._hasTreeinfo(fetcher, progresscb):
-            kernelpath = self._getTreeinfoMedia("kernel")
-            initrdpath = self._getTreeinfoMedia("initrd")
-        else:
+            try:
+                kernelpath = self._getTreeinfoMedia("kernel")
+                initrdpath = self._getTreeinfoMedia("initrd")
+            except ConfigParser.NoSectionError:
+                pass
+
+        if not kernelpath or not initrdpath:
             # fall back to old code
             if self.type is None or self.type == "hvm":
                 paths = self._hvm_kernel_paths
-- 
1.8.3.2

